### DISCUSSION TITLE
WinUI desktop UI tests

### DISCUSSION THREAD

**Discussion by adamplonka** [unoplatform/uno#7984] [General]
Are there any plans to support UI testing on WinUI3 desktop?

**Comment by jeromelaban** [comment]
We have plans to support this, though the testing landscape for WinUI 3 is still moving a lot and we have not settled on a specific technology to implement it in Uno.UITest.

**Comment by beargadin** [comment]
@jeromelaban Any update on this?

**Reply by jeromelaban** [reply]
Microsoft has still to provide plans for WinAppSDK testing, and we do not have an update to provide at this time.

**Reply by TopperDEL** [reply]
Just stumbled across this - what is the current state? I'm searching for a way to implement UITests for the WinUI-Uno-App (Windows). I do not get it to run with WinAppDriver but have absolutely no knowledge. So I might be doing many things wrong.
The documentation does not mention testing the WinUI-app either, does it?
Can you tell me if this is possible by now and where I might start? Thank you!

**Reply by jeromelaban** [reply]
At this time, UI testing for Uno is done through this library: https://github.com/unoplatform/uno.ui.runtimetests.engine

We're still working on getting further with this particular topic.
